src/main/java/ui/BoardAutoCreator.java

```diff
@@ -59,20 +61,38 @@ public BoardAutoCreator(UI ui, PanelControl panelControl, Preferences prefs) {
     public Menu generateBoardAutoCreateMenu() {
         Menu autoCreate = new Menu("Auto-create");
         MenuItem sample = new MenuItem(SAMPLE_BOARD);
-        sample.setOnAction(e -> createSampleBoard(true));
+        sample.setOnAction(e -> {
+            saveBoardAfterUserConfirmation(SAMPLE_BOARD);
+            createSampleBoard(true);
+        });
         autoCreate.getItems().add(sample);
 
         MenuItem milestone = new MenuItem(MILESTONES);
-        milestone.setOnAction(e -> createMilestoneBoard());
+        milestone.setOnAction(e -> {
+            saveBoardAfterUserConfirmation(MILESTONES);
+            createMilestoneBoard();
+        });
         autoCreate.getItems().add(milestone);
 
         MenuItem workAllocation = new MenuItem(WORK_ALLOCATION);
-        workAllocation.setOnAction(e -> createWorkAllocationBoard());
+        workAllocation.setOnAction(e -> {
+            saveBoardAfterUserConfirmation(WORK_ALLOCATION);
+            createWorkAllocationBoard();
+        });
         autoCreate.getItems().add(workAllocation);
 
         return autoCreate;
     }
 
+    private void saveBoardAfterUserConfirmation(String boardName) {
+        if (promptUserIfShouldSave(boardName)) ui.getMenuControl().saveBoardAs();
+    }
+
+    private boolean promptUserIfShouldSave(String boardName) {
```

Comment:
`isSaveBoardDialogResponsePositive` may read better in the call site. e.g. `if(isSaveBoardDialogResponsePositive(..)) ...`